src/Battlescape/Map.cpp

```diff
@@ -937,9 +937,10 @@ void Map::drawTerrain(Surface *surface)
 
 								for (int i = begin; i != end; i += direction)
 								{
-									tmpSurface = _projectileSet->getFrame(_projectile->getParticle(i));
-									if (tmpSurface)
+									Surface *ts = _projectileSet->getFrame(_projectile->getParticle(i));
```

Comment:
could you fix real problem not symptoms? This two lines are correct, where is error?

Reply:
Well, the getFrame returns  (Surface *)nullptr when no sprite defined. But the issue is the automatic dereference of nullptr awhen assigning a tmpSurface which is SurfaceRaw<const Uint8>... You mean then maybe change the return type of getFrame?

Reply:
Ok, right, error is on my part, small design hole, `SurfaceRaw` should accept `nullptr` without crash. And this is true error. Not this one line.

Reply:
I think I will fix it myself, thanks for spotting this bug.